src/main/java/nl/b3p/tailormap/api/geotools/featuresources/JDBCFeatureSourceHelper.java

```diff
@@ -5,64 +5,65 @@
+    if (tmfs.getAuthentication().getMethod() != ServiceAuthentication.MethodEnum.PASSWORD) {
+      throw new IllegalArgumentException(tmfs.getAuthentication().getMethod().getValue());
+    }
 
-  @SuppressWarnings("DoNotCallSuggester")
-  public static DataStore createDataStore(TMFeatureSource fs) throws IOException {
-    throw new UnsupportedEncodingException();
-    //    Map<String, Object> params = new HashMap<>();
-    //    JSONObject urlObj = new JSONObject(fs.getUrl());
-    //    params.put("dbtype", urlObj.get("dbtype"));
-    //    params.put("host", urlObj.get("host"));
-    //    params.put("port", urlObj.get("port"));
-    //    params.put("database", urlObj.get("database"));
-    //
-    //    params.put("schema", fs.schema);
-    //    params.put("user", fs.getUsername());
-    //    params.put(JDBCDataStoreFactory.FETCHSIZE.key, 50);
-    //    params.put("passwd", fs.getPassword());
-    //    params.put(JDBCDataStoreFactory.EXPOSE_PK.key, true);
-    //    params.put(JDBCDataStoreFactory.PK_METADATA_TABLE.key, "gt_pk_metadata");
-    //    // this key is available in ao. Oracle and MS SQL datastore factories, but not in the
-    // common
-    //    // parent..
-    //    // we need this for mssql to determine a featuretype on an empty table
-    //    if (!urlObj.get("dbtype").equals("oracle")) {
-    //      params.put(SQLServerDataStoreFactory.GEOMETRY_METADATA_TABLE.key, "geometry_columns");
-    //    }
-    //    Map<String, Object> logParams = new HashMap<>(params);
-    //    if (fs.getPassword() != null) {
-    //      logParams.put(
-    //          "passwd", String.valueOf(new char[fs.getPassword().length()]).replace("\0", "*"));
-    //    }
-    //    log.debug("Opening datastore using parameters: " + logParams);
-    //    DataStore ds = DataStoreFinder.getDataStore(params);
-    //
-    //    if (ds == null) {
-    //      throw new IOException("Cannot open datastore using parameters " + logParams);
-    //    }
-    //    return ds;
-  }
+    JDBCConnectionProperties c = tmfs.getJdbcConnection();
+    Objects.requireNonNull(c.getDbtype());
 
-  @Override
-  public SimpleFeatureSource openGeoToolsFeatureSource(
-      TMFeatureSource fs, TMFeatureType sft, int timeout) throws IOException {
-    return JDBCFeatureSourceHelper.openGeoToolsFSFeatureSource(fs, sft);
+    Map<String, Object> params = new HashMap<>();
+    params.put(DBTYPE.key, c.getDbtype().getValue());
+    params.put(HOST.key, c.getHost());
+    params.put(PORT.key, c.getPort() != null ? c.getPort() : defaultPorts.get(c.getDbtype()));
+    params.put(DATABASE.key, c.getDatabase());
+    params.put(SCHEMA.key, c.getSchema());
+    params.put(USER.key, tmfs.getAuthentication().getUsername());
+    params.put(PASSWD.key, tmfs.getAuthentication().getPassword());
+    params.put(FETCHSIZE.key, c.getFetchSize());
+    params.put(EXPOSE_PK.key, true);
+    params.put(PK_METADATA_TABLE.key, c.getPrimaryKeyMetadataTable());
```

Comment:
We should look into getting the table and column comments /remarks and add that to our service info.
This is new Geotools functionality, not yet released I think...

https://docs.geotools.org/latest/javadocs/org/geotools/data/oracle/OracleNGDataStoreFactory.html#GET_REMARKS